docker/pypi/msunmerged/Dockerfile

```diff
@@ -1,26 +1,54 @@
-FROM registry.cern.ch/cmsweb/gfal:latest as gfal
-FROM registry.cern.ch/cmsweb/dmwm-base:pypi-20240923-stable
-MAINTAINER Valentin Kuznetsov vkuznet@gmail.com
-COPY --from=gfal /data/miniconda /data/miniconda
+FROM registry.cern.ch/cmsweb/gfal:alma9-py3.12-v1.82.0 AS gfal2
+FROM registry.cern.ch/cmsweb/dmwm-base:alma9-py3.12-20250515
+LABEL org.opencontainers.image.authors="Alan Malta alan.malta@cern.ch, Dennis Lee dylee@fnal.gov"
+
+# Specific MSUnmerged requirements from epel repository
+RUN dnf install epel-release -y && dnf clean all && \
+    dnf -y install patch gfal2 python3-gfal2-util gfal2-plugin-http gfal2-plugin-dcap gfal2-plugin-file \
+                   gfal2-plugin-srm gfal2-plugin-xrootd gfal2-plugin-gridftp gfal2-plugin-sftp && \
+    dnf clean all
+
 ENV WDIR=/data
-ENV PATH $PATH:$WDIR/miniconda/bin
-ENV PYTHONPATH $WDIR/miniconda/lib/python3.8/site-packages/
+WORKDIR $WDIR
+
 # TAG to be passed at build time through `--build-arg TAG=<PYPI_TAG>`. Default: None
 ARG TAG=None
-WORKDIR $WDIR
-ADD run.sh $WDIR/run.sh
-# since we install gfal via external image we'll skip it for installation
-# of msunmerged, but to satisfy dependencies we'll install them first
+# gfal2 is built and copied from gfal2 stage, so we only install non-gfal2 service dependencies
+# FIXME: it is probably best to remove it from the requirements.txt file
 RUN curl -ksLO https://raw.githubusercontent.com/dmwm/WMCore/$TAG/requirements.txt
-RUN cat requirements.txt | grep -v gfal2 > req.txt
-RUN pip install -r req.txt
-RUN pip install --no-deps msunmerged==$TAG 
+RUN cat requirements.txt | grep dbs3-client > req.txt
```

Comment:
I don't think we need dbs3-client in this image, so I would suggest to completely remove this line grepping for it.
About L17 above, didn't you fix that FIXME message in the line below?